middleware/auth.js

```diff
@@ -4,54 +4,44 @@ const ValidationUtil = require('../util/validationUtil');
 
-const Auth = (req, res, next) => {
-    const token = req.headers['x-access-token'] || req.headers['authorization'];
 
-    // Decode token
-    if (token) {
-        // Remove unwanted prefix if exists
-        if (token.startsWith('Bearer ')) {
-            token = token.slice(7, token.length);
-        }
-        // Verify token
-        jwt.verify(token, AppConfig.SECRET, function (err, decoded) {
-            if (err) {
-                return res.status(HttpStatus.UNAUTHORIZED).json({ success: false, message: 'Unauthorized access.' });
-            } else {
-                req.decoded = decoded;
-                next();
-            }
-        });
-    } else {
-        return res.status(HttpStatus.FORBIDDEN).json({ success: false, message: 'No token provided.' });
-    }
-}
+const auth = (checkAccessType) => {
+    return (req, res, next) => {
+        const token = req.headers['x-access-token'] || req.headers['authorization'];
 
-const AdminAuth = (req, res, next) => {
-    const token = req.headers['x-access-token'] || req.headers['authorization'];
-
-    // Decode token
-    if (token) {
-        // Remove unwanted prefix if exists
-        if (token.startsWith('Bearer ')) {
-            token = token.slice(7, token.length);
-        }
-        // Verify token
-        jwt.verify(token, AppConfig.SECRET, function (err, decoded) {
-            // Only (super)admin is allowed to access this route
-            if (!err && ValidationUtil.isUserAdmin(decoded.accessType)) {
-                req.decoded = decoded;
-                next();
-            } else {
-                return res.status(HttpStatus.UNAUTHORIZED).json({ success: false, message: 'Unauthorized access.' });
+        // Decode token
+        if (token) {
+            // Remove unwanted prefix if exists
+            if (token.startsWith('Bearer ')) {
+                token = token.slice(7, token.length);
             }
-        });
-    } else {
-        return res.status(HttpStatus.FORBIDDEN).json({ success: false, message: 'No token provided.' });
+            // Verify token
+            jwt.verify(token, AppConfig.SECRET, function (err, decoded) {
+                if (!err) {
+                    // If checkAccessType is true => (super)admin-specific route
+                    if (checkAccessType && ValidationUtil.isUserAdmin(decoded.accessType)) {
+                        req.decoded = decoded;
+                        next();
+                    }
+                    // If checkAccessType is false => available for all logged users
+                    else if (!checkAccessType) {
+                        req.decoded = decoded;
+                        next();
+                    } else {
+                        // Logged user can't access (super)admin-specific routes
+                        return res.status(HttpStatus.UNAUTHORIZED).json({ success: false, message: 'Unauthorized access.' });
+                    }
```

Comment:
I still think you should have done it as discussed in a previous comment, since that way would give us more flexibility if we are to make another app for internal use and another user-type for us (super-super-admin). It would make it a lot easier to add that functionality, plus we would have a clearer understanding which function does what: `authUser`, `authAdmin` are more understandable than `auth` and `auth(true)`. Plus if we are to add another user group for us we would just call `authSuperAdmin` (or whatever) whilst here we would have to make another enum or something like that.

Reply:
Yea, for now it is okay but another approach would be to make the parameter userType, for example auth(UserTypeEnum['Admin']) would be one way for calling the function to authenticate it for the Admin type. auth function would do the authentication based on the enum check.